### DISCUSSION TITLE
Dispatcher stops working with hanging in NtUserGetMessage

### DISCUSSION THREAD

**Discussion by BenSchoen** [dotnet/wpf#5605] [Q&A]
- .NET 4.8
- Windows 10 Enterprise LTS 1809
- WPF

Problem:
Our wpf app sometimes freezes and everything is "disabled". The issue is very random and not reproducable. Usually it happens once a day and not every time in the same context. The dispatcher is on executing NtUserGetMessage and there is no usercode blocking the dispatcher. Installing a newer windows version made the issue rarer. The application background stuff is working further.

I am unsure if its really a bug inside wpf. Are there any suggestions to debug the issue further or is something similar known? 

![image](https://user-images.githubusercontent.com/93266740/139076437-808456f3-bc7b-4093-b3fa-d96b036a38b8.png)

**Comment by lindexi** [comment]
It stops in NtUserGetMessage as expected. WPF application will receive the windows message by GetMessage. It is stop in GetMessage when the application is idle.

**Comment by BenSchoen** [comment]
Ok, got it. What could be the the reason for a not responsive visual. In this state of the callstack all buttons are disabled, including those, which are enabled all the time.

**Comment by vishalmsft** [comment]
@BenSchoen As this is a question, I am moving this to Discussions->Q&A. 
As you said, installing new version of Windows made the issue rarer, hung issue may be due to other reason. 
Have your tried it on more than one machine?
